### DISCUSSION TITLE
stalledInterval doesn't work

### DISCUSSION THREAD

**Discussion by sagiassa** [OptimalBits/bull#2234] [Q&A]
Hi, 
I am trying to use stalledInterval settings to prevent from a process to be stalled in the default time. 
It seems like nothing I do affect the job to get to on stalled.

Can I get an example of a function that does work?

**Comment by manast** [comment]
Normally you do not need to modify this parameter, so the question is how are your process functions implemented so that they end stalling jobs in the first place?

**Reply by sagiassa** [reply]
Well, I'm trying to go over a big array and register another job in batches. But it seems that the loop doesn't end and the job is stalled.
From your answer I understand that I have to change something in my functionality and not in the bull settings, right?

**Comment by manast** [comment]
In general, keeping the event loop busy in nodejs is not a good thing. You can break the array iteration into small chunks and return the code execution to the event loop in between using await for example.

**Comment by sibelius** [comment]
did you figure it out?

**Comment by manast** [comment]
Not sure what you mean by that, I always “bother” to explain everything with the best of my abilities, I am not infalible though. Here is one of the places where stalledInterval is explained: https://docs.bullmq.io/guide/workers/stalled-jobs <https://docs.bullmq.io/guide/workers/stalled-jobs> 


On 10 Mar 2025, at 12:29, Christian ***@***.***> wrote:


Everytime someone asks for advice on how to set this parameter, you're always telling them that they shouldn't need to touch it and never actually offer guidance. Why even make it an option if you're never going to bother explaining it then?
